core/src/main/scala/org/genivi/sota/core/db/Campaigns.scala

```diff
@@ -22,27 +23,37 @@ object Campaigns {
   import Campaign._
   import SotaCoreErrors._
 
+  private def optionPackageId(name: Option[PackageId.Name], version: Option[PackageId.Version]): Option[PackageId] =
+    if(name.isDefined && version.isDefined) {
+      Some(PackageId(name.get, version.get))
+    } else {
+      None
+    }
```

Comment:
```
(name, version) match {
  case (Some(name), Some(version)) => Some(PackageId(name, version)
  case _ => None
}
```

I'm sure scala have some fancy applicative syntax one could use, but I don't know it so the above is good enough.